src/components/formik-santiment-ui/FormikCheckboxes.js

```diff
@@ -8,25 +8,42 @@ const FormikCheckboxes = ({
   name,
   disabled = false,
   styles
-}) => (
-  <Field
-    name={name}
-    render={({ field, form }) => (
-      <Checkboxes
-        options={options}
-        disabledIndexes={disabledIndexes}
-        defaultSelectedIndexes={field.value}
-        onSelect={value => {
-          const newData = field.value.find(el => el === value)
-            ? field.value.filter(el => el === name)
-            : [...field.value, value]
-          form.setFieldValue(name, newData)
-          form.setFieldTouched(name, true)
-        }}
-        style={styles}
-      />
-    )}
-  />
-)
+}) => {
+  return (
+    <Field
+      name={name}
+      render={({ field, form }) => {
+        let preselectedValues = []
+
+        if (disabledIndexes && disabledIndexes.length > 0) {
+          preselectedValues = options.filter(
+            x => !disabledIndexes.find(y => y === x)
+          )
+        }
+
+        if (!field.value) {
+          form.setFieldValue(name, preselectedValues)
+          field.value = preselectedValues
+        }
+
+        return (
+          <Checkboxes
+            options={options}
+            disabledIndexes={disabledIndexes}
+            defaultSelectedIndexes={field.value}
+            onSelect={value => {
```

Comment:
`onSelect` returns all selected indexes as [the second argument](https://github.com/santiment/san-ui/blob/b51d26b17babaf4f45d4ce0a2abb07317a50dd87/src/Selects/BaseSelect.js#L53). Can it be used instead of custom filtering/array appending?

Reply:
Correct, fixed.

Reply:
If it's correct, why the fix was not applied?